impl_core/src/main/java/io/opencensus/implcore/tags/propagation/CorrelationContextFormat.java

```diff
@@ -0,0 +1,175 @@
+      int totalChars = 0; // Here chars are equivalent to bytes, since we're using ascii chars.
+      int totalTags = 0;
+      for (Iterator<Tag> i = InternalUtils.getTags(tagContext); i.hasNext(); ) {
+        Tag tag = i.next();
+        if (TagTtl.NO_PROPAGATION.equals(tag.getTagMetadata().getTagTtl())) {
+          continue;
+        }
+        if (stringBuilder.length() > 0) {
+          stringBuilder.append(TAG_DELIMITER);
+        }
+        totalTags++;
+        totalChars += encodeTag(tag, stringBuilder);
+      }
+      checkArgument(
+          totalTags <= MAX_NUMBER_OF_TAGS,
+          "Number of tags in the TagContext exceeds limit " + MAX_NUMBER_OF_TAGS);
+      // Note per W3C spec, only the length of tag key and value counts towards the total length.
+      // Length of properties (a.k.a TagMetadata) does not count.
+      checkArgument(
+          totalChars <= TAGCONTEXT_SERIALIZED_SIZE_LIMIT,
+          "Size of TagContext exceeds the maximum serialized size "
+              + TAGCONTEXT_SERIALIZED_SIZE_LIMIT);
+      setter.put(carrier, CORRELATION_CONTEXT, stringBuilder.toString());
+    } catch (IllegalArgumentException e) {
+      throw new TagContextSerializationException("Failed to serialize TagContext", e);
+    }
+  }
+
+  // Encodes the tag to the given string builder, and returns the length of encoded key-value pair.
+  private static int encodeTag(Tag tag, StringBuilder stringBuilder) {
+    String key = tag.getKey().getName();
+    String value = tag.getValue().asString();
+    int charsOfTag = key.length() + value.length();
+    // This should never happen with our current constraints (<= 255 chars) on tags.
+    checkArgument(
+        charsOfTag <= TAG_SERIALIZED_SIZE_LIMIT,
+        "Serialized size of tag " + tag + " exceeds limit " + TAG_SERIALIZED_SIZE_LIMIT);
+
+    // TODO(songy23): do we want to encode TagMetadata?
+    stringBuilder.append(key).append(TAG_KEY_VALUE_DELIMITER).append(value);
+    return charsOfTag;
+  }
+
+  @Override
+  public <C /*>>> extends @NonNull Object*/> TagContext extract(C carrier, Getter<C> getter)
+      throws TagContextDeserializationException {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+    if (State.DISABLED.equals(state.getInternal())) {
+      return TagMapImpl.EMPTY;
+    }
+
+    @Nullable String correlationContext = getter.get(carrier, CORRELATION_CONTEXT);
+    if (correlationContext == null) {
+      throw new TagContextDeserializationException(CORRELATION_CONTEXT + " not present.");
+    }
+    try {
+      if (correlationContext.isEmpty()) {
+        return TagMapImpl.EMPTY;
+      }
+      Map<TagKey, TagValueWithMetadata> tags = new HashMap<>();
+      List<String> stringTags = TAG_DELIMITER_SPLITTER.splitToList(correlationContext);
+      for (String stringTag : stringTags) {
+        List<String> keyValuePair = TAG_KEY_VALUE_DELIMITER_SPLITTER.splitToList(stringTag);
```

Comment:
before key and value is split, property list should be split because property list may also contain '=' delimiter.

Reply:
Good catch, fixed.